include/GIL/Type.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef GLU_GIL_TYPE_HPP
+#define GLU_GIL_TYPE_HPP
+
+#include "AST/Types/TypeBase.hpp"
+
+namespace glu::gil {
+
+/// @brief Represents a type in the GIL (Glu Intermediate Language).
+/// For more information, see the documentation here:
+/// https://glu-lang.org/gil/
+class Type {
+    /// @brief struct to store the fields of the Type class.
+    ///        It has 10 bytes that are unused.
+    struct Fields {
+        uint64_t size : 48;
+        uint64_t alignment : 5;
+        uint64_t isConst : 1;
+    } _fields;
+    glu::types::TypeBase *_type;
+
+public:
+    /// @brief Constructor for the Type class.
+    /// @param size The size of the type in bytes.
+    /// @param alignment The alignment of the type in bytes.
+    /// @param isConst Whether the type is const or not.
+    /// @param type A pointer to the type base.
+    Type(
+        size_t size, size_t alignment, bool isConst, glu::types::TypeBase *type
+    )
+        : _type(type)
+    {
+        _fields.size = size;
+        _fields.alignment = alignment;
+        _fields.isConst = isConst;
```

Comment:
can't you use an initializer list?
```suggestion
        : _type(type)
        , _fields { size, alignment, isConst }
    {
```

Reply:
You can also add assertions verifying that size and alignments weren't truncated by the bit field:
```c++
assert(_fields.size == size && "Size is bigger than 48 bits!")
```